### DISCUSSION TITLE
Custom column header / context menu

### DISCUSSION THREAD

**Discussion by danisal** [nick-keller/react-datasheet-grid#83] [Q&A] [answered]
Is it possible to add custom column headers? or even a context menu?

I see that to add a specific component for each column, that will render on each cell, but not for the column header, as I would like to add a different behavior/context menu for some columns...

The `contextMenuComponent` function doesn't give me where it was called, e.g. which column, so I could implement the custom column context menu...

My idea is to pass a component in the title property... not sure if this is the best approach tho...

**Comment by nick-keller** [comment, marked as answer]
Hey @danisal !
Just pass your header component to the [title](https://react-datasheet-grid.netlify.app/docs/api-reference/columns#title) prop of the column:

```jsx
const columns = [
  { title: <MyComponent /> }
]
```

Now regarding the context menu, it is true that you do not have access to the selected area. Please create a feature request detailing your use case precisely to help scope the feature and prioritize the roadmap!
